Dockerfile

```diff
@@ -5,7 +5,9 @@ WORKDIR /build
 # install dependencies and go 1.16
 
 # copy just enough of the git repo to parse HEAD, used to record version in OLM binaries
-RUN dnf update -y && dnf install -y bash make git mercurial jq wget golang && dnf upgrade -y
+RUN dnf update -y && dnf install -y bash make git mercurial jq wget && dnf upgrade -y
+RUN curl -sSL https://go.dev/dl/go1.17.6.linux-amd64.tar.gz | tar -xzf - -C /usr/local
+ENV PATH=/usr/local/go/bin:$PATH
```

Comment:
Hmm this feels fairly hacky - it might be time to reword the build image that we use for this and work towards using a proper Go image. It's unclear to me looking at this stage why we need a fedora base image, as we're just building the OLM binaries and any utility packages needed in the pipeline.

Reply:
I would not disagree that it would make sense to have a better for purpose base image. As I was focusing on the version upgrade I kept the existing approach: download and unpack

Reply:
Yeah, totally fair, that was more of a PSA than me requesting changes. Maybe we should be tracking this as an upstream issue then?

Just a quick note to other reviewers, it looks like these changes are needed as [builds were failing](https://github.com/operator-framework/operator-lifecycle-manager/pull/2574#issuecomment-1016368075) as this Dockerfile uses a fedora image that doesn't contain an up-to-date Go minor version.